core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -313,9 +316,11 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
         final Channel channel = ctx.channel();
         final RequestHeaders headers = req.headers();
         final String hostname = hostname(headers);
-        final VirtualHost virtualHost = config.findVirtualHost(hostname);
+        final VirtualHost virtualHost = configHolder.getConfig().findVirtualHost(hostname);
```

Comment:
Could we call `configHolder.getConfig()` only once for each request, so that a request is handled with one single `ServerConfig` for consistency's sake? Let's declare a local variable and pass it as a parameter where necessary.

Reply:
Makes sense. I will make the change.